core/src/util.jl

```diff
@@ -348,13 +354,18 @@ Function that goes smoothly from 0 to 1 in the interval [0,threshold],
 and is constant outside this interval.
 """
 function reduction_factor(x::T, threshold::Real)::T where {T <: Real}
-    return if x < 0
-        zero(T)
-    elseif x < threshold
-        x_scaled = x / threshold
-        (-2 * x_scaled + 3) * x_scaled^2
+    if isinf(x)
+        (sign(x) + one(T)) / 2
     else
-        one(T)
+        x_scaled = x / threshold
+        0.5 * (
+            0.5 *
+            (sign(threshold - x) + one(T)) *
+            (sign(x) + one(T)) *
+            ((-2 * x_scaled + 3) * x_scaled^2) +
+            sign(x - threshold) +
+            one(T)
+        )
     end
 end
```

Comment:
When is the input infinite? Can you add a test for infinite input in the "reduction_factor" testitem?

Reply:
The input can be infinite because of certain parameters that default to `+/Inf` (e.g. reference levels).